.github/workflows/build-devcontainer.yml

```diff
@@ -0,0 +1,37 @@
+name: Build and publish devcontainer
+
+on:
+  pull_request:
```

Comment:
I'm confused what are you trying to achieved with this? Because putting it this way, any merged pull request that has the path will trigger a new devcontainer build.

If we're making it only trigger on merged pull request to master, we can change this to use:

```yaml
on:
  push:
    branches:
      - master
    paths:
      - .devcontainer/Dockerfile
```

and then lock push access to the master branch, change the config to only allow push to Master via a PR.

Reply:
Yeah will fix. It's been awhile since I am on Github Actions. The intention behind it is only when it's pull and merged, it will trigger the build. As such the `on` path is combined with:
```
if: github.event.pull_request.merged == true
```

per the documentation here: https://docs.github.com/en/actions/using-workflows/events-that-trigger-workflows#running-your-pull_request-workflow-when-a-pull-request-merges

However, now that I think about it. Doing it this way means if the PR is to a `development` branch for example, it will trigger a rebuild of a devcontainer as well and that is a non-intentional behaviour.

Reply:
yeah I _feel_ like the desired behavior is to only trigger this when a PR is merged to master, so if that's the case just adjust accordingly.